nearcore/src/config.rs

```diff
@@ -1327,39 +1336,98 @@ pub fn load_config(
     dir: &Path,
     genesis_validation: GenesisValidationMode,
 ) -> anyhow::Result<NearConfig> {
-    let config = Config::from_file(&dir.join(CONFIG_FILENAME))?;
-    let genesis_file = dir.join(&config.genesis_file);
+    let genesis_validation = match genesis_validation {
+        GenesisValidationMode::Full => GenesisValidationMode::Full,
+        GenesisValidationMode::UnsafeFast => GenesisValidationMode::UnsafeFast,
+    };
+    let mut validation_errors = ValidationErrors::new();
+
+    // if config.json has file issues, the program will directly panic
+    let config = Config::from_file_skip_validation(&dir.join(CONFIG_FILENAME))?;
+    // do config.json validation later so that genesis_file, validator_file and genesis_file can be validated before program panic
+    config.validate().map_or_else(|e| validation_errors.push_errors(e), |_| ());
+
     let validator_file = dir.join(&config.validator_key_file);
     let validator_signer = if validator_file.exists() {
-        let signer = InMemoryValidatorSigner::from_file(&validator_file).with_context(|| {
-            format!("Failed initializing validator signer from {}", validator_file.display())
-        })?;
-        Some(Arc::new(signer) as Arc<dyn ValidatorSigner>)
+        match InMemoryValidatorSigner::from_file(&validator_file) {
+            Ok(signer) => Some(Arc::new(signer) as Arc<dyn ValidatorSigner>),
+            Err(_) => {
+                validation_errors.push_errors(ValidationError::ValidatorKeyFileError {
+                    error_message: format!(
+                        "Failed initializing validator signer from {}",
+                        validator_file.display()
+                    ),
+                });
+                None
+            }
+        }
     } else {
         None
     };
+
     let node_key_path = dir.join(&config.node_key_file);
-    let network_signer = NodeKeyFile::from_file(&node_key_path).with_context(|| {
-        format!("Failed reading node key file from {}", node_key_path.display())
-    })?;
+    let network_signer_result = NodeKeyFile::from_file(&node_key_path);
+    let network_signer = match network_signer_result {
+        Ok(node_key_file) => Some(node_key_file),
+        Err(_) => {
+            validation_errors.push_errors(ValidationError::NodeKeyFileError {
```

Comment:
with the `ValidationError` changed to `ReadConfigError`, the name should make more sense. 

> If there are problems reading the basic files involved, then we just print that error out and only after you've set things up properly on a basic level will you hear about validation errors.

I assume by validation here it means checking whether the fields' values are incorrect. There's actually no validation on fields of validator_key or network_signer, validations only happen for genesis.json and config.json. If the file itself cannot be de-serialized into the corresponding config object, then we will push the error and won't do validation for this file. Validation only happens when the config file de-serialize with no error.

Reply:
will leave it as is for now.

Reply:
right, I just meant that it should be possible to just leave this part as it was before, since the old logic would probably still be just fine and it would tighten the scope of this PR.